src/filter/http_context.rs

```diff
@@ -207,6 +214,15 @@ impl HttpContext for Filter {
     fn on_http_request_headers(&mut self, _: usize, _: bool) -> Action {
         info!("on_http_request_headers #{}", self.context_id);
 
+        let req_headers = self.get_http_request_headers();
+        for (header, value) in req_headers.iter() {
+            match header.as_str() {
+                "traceparent" | "tracestate" | "baggage" => {
```

Comment:
`self.get_http_request_headers` returns lowercase header names? in other words, if the header is `TraceParent`, is the value captured? According to https://w3c.github.io/trace-context/#traceparent-header

```
The header name is [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive). That is, TRACEPARENT, TraceParent, and traceparent are considered the same header. The header name is a single word; it does not contain any delimiters such as a hyphen.
```

Reply:
Yep from what I could tell no matter the provided format the headers were retrieved in lowercase - perhaps due to this https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/header_casing

I could add a `.to_lowercase()` before the match for safety in case though. Also considering moving the three header strings out to constants

Reply:
if it's only used there, no need for a constant.